src/jsx/import/input-data/supply-dataset-file.jsx

```diff
@@ -9,85 +9,82 @@
  * Note: this script implements jsx (reactjs) syntax.
  */
 
-import React from 'react';
+import React, { Component } from 'react';
 import checkValidFile from './../validator/valid-file.js';
 
-var SupplyDatasetFile = React.createClass({
-  // initial 'state properties'
-    getInitialState: function() {
+class SupplyDatasetFile extends Component {
+    // initial 'state properties'
+    getInitialState() {
         return {
-            additional_elements: []
+            additional_elements: [],
         };
-    },
-  // update 'state properties': index for additional input elements
-    handleAddMore: function(event){
-        var elements = this.state.additional_elements;
+    }
+    // update 'state properties': index for additional input elements
+    handleAddMore(event) {
+        const elements = this.state.additional_elements;
         elements.push(true);
-        this.setState({additional_elements: elements});
-        this.props.onChange({submitted_proper_predictor: false});
-    },
-    handleRemove: function(event){
-        var elements = this.state.additional_elements;
-        var datasetBoolean = true;
+        this.setState({ additional_elements: elements });
+        this.props.onChange({ submitted_proper_predictor: false });
+    }
+    handleRemove(event) {
+        const elements = this.state.additional_elements;
+        let datasetBoolean = true;
```

Comment:
I like how you use `let` here. I'm often lazy, and use `var` everywhere, and sometimes use `const`, when I remember.

Reply:
I think it's really important to get rid of vars, because it hoists and does not have block scope 
But it's not so easy, maybe I will open an issue for it.